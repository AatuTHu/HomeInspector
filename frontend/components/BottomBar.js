import { Text, View,TouchableOpacity } from 'react-native';
import { styles } from '../styles/styles';
import { buttons } from '../styles/buttonStyles';

export default function BottomBar({setScreen}) {
  return (
    <View style = {styles.navContainer}>
      <TouchableOpacity style = { buttons.navButtons } onPress={ () => setScreen(3) } >
        <View>
          <Text>
            Notes
          </Text>
        </View>
      </TouchableOpacity>
      
      <TouchableOpacity style = { buttons.navButtons } onPress={ () => setScreen(1) } >
        <View>
          <Text>
            Home
          </Text>
        </View>
      </TouchableOpacity>

      <TouchableOpacity style = { buttons.navButtons } onPress={ () => setScreen(2) } >
        <View>
          <Text>
            Profile
          </Text>
        </View>
      </TouchableOpacity>
    </View>
  )
}