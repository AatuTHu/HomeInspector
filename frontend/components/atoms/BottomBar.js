import { Text, View,TouchableOpacity } from 'react-native';
import { styles } from '../../styles/styles';
import { buttons } from '../../styles/buttonStyles';
import  Icon1 from 'react-native-vector-icons/Feather'
import Icon2 from 'react-native-vector-icons/AntDesign'

export default function BottomBar({setScreen}) {
  return (
    <View style = {styles.navContainer}>
      <TouchableOpacity style = { buttons.navButtons } onPress={ () => setScreen(3) } >
        <Text>
          <Icon2 name="pushpino" size={25}/>
        </Text>
      </TouchableOpacity>
      
      <TouchableOpacity style = { buttons.navButtons } onPress={ () => setScreen(1) } >
        <Text>
          <Icon1 name="home" size={25}/>
        </Text>
      </TouchableOpacity>
      
      <TouchableOpacity style = { buttons.navButtons } onPress={ () => setScreen(2) } >
        <Text>
          <Icon2 name="dashboard" size={25}/>
        </Text>
      </TouchableOpacity>
    </View>
  )
}