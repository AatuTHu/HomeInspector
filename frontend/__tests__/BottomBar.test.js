import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import BottomBar from '../components/atoms/BottomBar';

describe('BottomBar', () => {
  it('renders correctly with custom props', () => {
    const screen = render(<BottomBar />);
    expect(screen).toMatchSnapshot();
  });

  it('Switches screens', () => {
    
    const mockFunction = jest.fn();

    const { getByText } = render(<BottomBar setScreen = {mockFunction}/>);
  
    const screen1 = getByText('Notes');
    const screen2 = getByText('Home');
    const screen3 = getByText('Control');

    fireEvent.press(screen1);
    fireEvent.press(screen2);
    fireEvent.press(screen3);

    expect(mockFunction).toBeCalled();

  })
});
