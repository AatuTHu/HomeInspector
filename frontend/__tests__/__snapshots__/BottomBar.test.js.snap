// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BottomBar renders correctly with custom props 1`] = `
<View
  style={
    {
      "display": "flex",
      "flexDirection": "row",
      "justifyContent": "space-around",
      "width": "100%",
    }
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "borderRadius": 5,
        "height": "100%",
        "justifyContent": "center",
        "opacity": 1,
        "width": "33%",
      }
    }
  >
    <Text>
      Notes
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "borderRadius": 5,
        "height": "100%",
        "justifyContent": "center",
        "opacity": 1,
        "width": "33%",
      }
    }
  >
    <Text>
      Home
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "borderRadius": 5,
        "height": "100%",
        "justifyContent": "center",
        "opacity": 1,
        "width": "33%",
      }
    }
  >
    <Text>
      Control
    </Text>
  </View>
</View>
`;
